mangohud support: MangoHud::getLibraryString should return absolute path

cthbleachbit opened the pull request:
Some distros ship MangoHub vulkan layer json with bare shared object name instead of an absolute path. This breaks environment config as `MinecraftInstance::createLaunchEnvironment()` seems to require absolute path of `libMangoHud.so`.

Since we already have `MangoHud::findLibrary()` lying around, use that to figure out where `libMangoHud.so` really is.

Also made a few related variables into const-reference along the way.

Trial97 commented on launcher/MangoHud.cpp:
```diff
@@ -111,10 +112,18 @@ QString getLibraryString()
+            QString libraryPath = findLibrary(libraryName);
+            if (!libraryPath.isEmpty()) {
+                return libraryPath;
+            }
+        }
```
Wouldn't it be best to just return the libraryName if the findLibrary fails to find it?
That way we are consistent with old behavior.
```suggestion
            }
            return libraryName;
        }
```

cthbleachbit replied:
`findLibrary()` also doubles as a check that this shared object indeed can be dlopen and loaded into a process.

Not sure what merit this would have to return a string when the library is not useable.

Trial97 replied:
maybe I misunderstood but `findLibrary` only works for glibc and not clang.
So in case of clang the dlopen will not happen and `findLibrary` will return an empty string.

My only requirement here is in case `findLibrary` fails in any way to return back the `libraryName` like we used to until now

cthbleachbit replied:
Ah I see your concerns now - dlopen would only work on linux based systems (afaik not related to specific compiler, whether that's gcc or clang. It will work as long as system C library supports `dlopen()`, whether that's glibc or musl or something).

Will push an amend shortly.

cthbleachbit committed:
mangohud support: getLibraryString should return absolute path when possible

Some distros ship MangoHub vulkan layer json with bare shared object name
instead of an absolute path. This breaks environment config as
`MinecraftInstance::createLaunchEnvironment()` seems to require absolute
path of `libMangoHud.so`.

Since we already have `MangoHud::findLibrary()` lying around, use that to
figure out where `libMangoHud.so` really is. In case of a platform that
doesn't support `dlopen()`, fallback to old behavior and return the path
verbatim as it is recorded in vk layer json.

Signed-off-by: Tianhao Chai <cth451@gmail.com>
